search-solr/src/main/java/edu/unc/lib/boxc/search/solr/services/SearchStateFactory.java

```diff
@@ -397,13 +397,35 @@ private void populateSearchState(SearchState searchState, Map<String,String[]> r
                 }
             }
             searchState.setFacetsToRetrieve(facetsToRetrieve);
+
+
+            //Add date added year range, which is a sort of facet
+            try {
+                SearchState.RangePair dateCreatedYear = new SearchState.RangePair();
+                parameter = getParameter(request, searchSettings.searchFieldParam(
+                        SearchFieldKey.DATE_CREATED_YEAR.name()));
+                if (parameter != null && parameter.length() > 0) {
+                    String[] dateRange = parameter.split(",");
```

Comment:
I'd suggest using the `RangePair(String pairString)` constructor here since it already knows how to parse the range field syntax, and adding a call inside of the constructor to `checkValidRangePair` to make sure its valid.